### ISSUE TITLE
Compilation failure with GCC 8.4.1

### ISSUE THREAD

**Issue by robertu94** [ornladios/ADIOS2#3223]
**Describe the bug**
When building with spack and gcc 8.4.1, build fails because adios2 includes the `<filesystem>` after checking the appropriate ` __cpp_lib_filesystem ` macro, but doesn't link the corresponding libraries or pass the appropriate standards flag.  The current method used in cmake to enforce c++11 doesn't, it ensure *at least* C++11

**To Reproduce**
spack install adios2%gcc@8.4.1

**Expected behavior**
The build succeeds

**Desktop (please complete the following information):**
 - OS/Platform: Centos 8.5.2111 (Palmetto Cluster at Clemson University)
 - Build [e.g. compiler version gcc 8.4.1, cmake version 3.23, build type: shared ]

**Additional context**
A collaborator at Clemson encountered this failure [Build log with failure](https://files.slack.com/files-pri/TN503Q6KF-F03GEBV1FPB/logbuild.txt)

**Following up**
N/A

**Comment by robertu94** [comment]
A copy of the file on Github in case slack deletes it [logbuild.txt](https://github.com/ornladios/ADIOS2/files/8734539/logbuild.txt)

**Comment by williamfgc** [comment]
@robertu94 thanks for reporting this, you're right as adios2 should be compiling strictly with C++11. Pinging @chuckatkins as it seems to be on the CMake side handling gcc8 requiring extra libraries. That should go away in gcc9 if it helps.

**Comment by robertu94** [comment]
@williamfgc thanks for the quick response. I know that gcc9 fixes the issue, but we were trying to use the system compiler where possible since it simplifies working with MPI.

**Comment by williamfgc** [comment]
@robertu94 makes sense, in that case spack should be setup to reuse the system compiler and not build gcc from source. If that's the case, might be worth a shot using gcc9 so you won't get stuck.

**Comment by robertu94** [comment]
Closing because we no longer need GCC 8 support.
